src/Processors/Transforms/Streaming/UserDefinedEmitStrategyAggregatingTransform.cpp

```diff
@@ -41,25 +42,16 @@ void UserDefinedEmitStrategyAggregatingTransform::finalize(const ChunkContextPtr
     if (variants.empty())
         return;
 
-    Block block;
-    if (params->final)
-    {
-        auto results = params->aggregator.convertToBlocksFinal(variants, ConvertAction::STREAMING_EMIT, 1);
-        assert(results.size() == 1);
-        block = std::move(results.back());
-
-        if (params->emit_version)
-            emitVersion(block);
-    }
+    Chunk chunk;
+    if (variants.type == AggregatedDataVariants::Type::without_key)
+        chunk = AggregatingHelper::convertWithoutKey(variants, *params);
     else
-    {
-        auto results = params->aggregator.convertToBlocksIntermediate(variants, ConvertAction::STREAMING_EMIT, 1);
-        assert(results.size() == 1);
-        block = std::move(results.back());
-    }
+        chunk = AggregatingHelper::convertSingleLevel(variants, *params);
+
+    if (params->emit_version && params->final)
+        emitVersion(chunk);
 
-    if (block)
-        setCurrentChunk(convertToChunk(block), chunk_ctx);
+    setCurrentChunk(std::move(chunk), chunk_ctx);
```

Comment:
if UDA does not emit, you should not set empty chunk and emit it.

Reply:
in fact, I have checked in setCurrentChunk()